jobmaster/dm/checkpoint/agent.go

```diff
@@ -0,0 +1,206 @@
+package checkpoint
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/hanfei1991/microcosm/jobmaster/dm/config"
+	"github.com/hanfei1991/microcosm/jobmaster/dm/metadata"
+	"github.com/hanfei1991/microcosm/lib"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	dmconfig "github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/dm/pkg/conn"
+	"github.com/pingcap/tiflow/dm/pkg/cputil"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/zap"
+)
+
+const (
+	loadCheckpointTable = `CREATE TABLE IF NOT EXISTS ? (
```

Comment:
I'm not sure this `?` can be used to represent identifiers :thinking: 

> Parameter markers can be used only where data values should appear, not for SQL keywords, identifiers, and so forth.

https://dev.mysql.com/doc/refman/5.7/en/prepare.html